livre/Eq/Window.cpp

```diff
@@ -16,19 +18,185 @@
  * along with this library; if not, write to the Free Software Foundation, Inc.,
  * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
  */
-#include <eq/gl.h>
 
 #include <livre/Eq/Window.h>
+
 #include <livre/Eq/Error.h>
+#include <livre/Eq/FrameData.h>
+#include <livre/Eq/Node.h>
+#include <livre/Eq/Pipe.h>
+#include <livre/Eq/Render/EqContext.h>
+
+#include <livre/core/Dash/DashTree.h>
+#include <livre/core/DashPipeline/DashConnection.h>
+#include <livre/core/DashPipeline/DashProcessor.h>
+#include <livre/core/DashPipeline/DashProcessorInput.h>
+#include <livre/core/DashPipeline/DashProcessorOutput.h>
+
+#include <livre/Lib/Configuration/VolumeRendererParameters.h>
 
+#include <livre/Lib/Uploaders/DataUploadProcessor.h>
+#include <livre/Lib/Uploaders/TextureUploadProcessor.h>
+
+#include <eq/gl.h>
+
+#define CONNECTION_ID 0u
+const uint32_t maxQueueSize = 65536;
 
 namespace livre
 {
 
+class Window::Impl
+{
+public:
+    Impl( Window* window )
+        : _window( window )
+        , _dashProcessorPtr( new DashProcessor( ))
+    {}
+
+    void configInitGL()
+    {
+        const Window* sharedWindow = static_cast< const Window* >( _window->getSharedContextWindow( ));
+        if( !sharedWindow || sharedWindow == _window )
+        {
+            _windowContext.reset( new EqContext( _window ));
+            initializePipelineProcessors();
+            initializePipelineConnections();
+        }
+        else
+        {
+            _windowContext = sharedWindow->_impl->_windowContext;
+            _dashProcessorPtr = sharedWindow->_impl->_dashProcessorPtr;
+            _textureUploadProcessorPtr = sharedWindow->_impl->_textureUploadProcessorPtr;
+            _dataUploadProcessorPtr = sharedWindow->_impl->_dataUploadProcessorPtr;
+        }
+    }
+
+    void configExitGL()
+    {
+        if( !_dashProcessorPtr.unique( ))
```

Comment:
I should use texture- or dataUploadProcessor here. dashProcessor is not reset() in exit of window....